### PR TITLE
Fix squeeze bug for length 1 array, and inconsistency on requirements and setup.py 

### PR DESCRIPTION
**Pull request by brahmaneya**
#Description of proposed changes
Squeezing a length 1 array for used to create a scalar instead of a 1D array as intended. Fixed by adding an extra check.
Also made munkres version in setup.py consistent with requirements.

## Test plan
Update unit test for changed function.

## Checklist

Need help on these? Just ask!

* [x] I have read the **CONTRIBUTING** document.
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] I have run `tox -e complex` and/or `tox -e spark` if appropriate.
* [ ] All new and existing tests passed.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -35,7 +35,7 @@
     packages=find_packages(),
     include_package_data=True,
     install_requires=[
-        "munkres==1.1.2",
+        "munkres>=1.0.6",
```
Comment (vincentschen) [inline setup.py]: are these version commits related to this patch?

File: test/utils/test_core.py [hunk 2]
```diff
@@ -21,6 +21,10 @@ def test_to_int_label_array(self):
         Y = to_int_label_array(X, flatten_vector=True)
         np.testing.assert_array_equal(Y, Y_expected)
 
+        Y = to_int_label_array(np.array([[1]]), flatten_vector=True)
+        Y_expected = np.array([1])
+        np.testing.assert_array_equal(Y, Y_expected)
+
```
Comment (vincentschen) [inline test/utils/test_core.py]: nice!

File: setup.py (hunk 1 above)
Comment (brahmaneya) [reply setup.py]: They're not, but `tox -e check` was failing without it.

Comment (brahmaneya) [reply setup.py]: I'll try reverting it and see if travis ci passes.

Comment (brahmaneya) [reply setup.py]: CI fails if I revert.
